extension-manifest-v3/src/background/adblocker.js

```diff
@@ -326,7 +339,10 @@ if (__PLATFORM__ === 'safari') {
   });
 } else {
   chrome.webNavigation.onCommitted.addListener(async (details) => {
-    injectScriptlets(details.tabId, details.url);
+    const isMainFrame = details.parentFrameId === -1;
+    if (isMainFrame || !(await isPausedTab(details.tabId))) {
```

Comment:
The `details` object contains the `url`. Why do you need to use `chrome.tabs` API? We can simply call `isPaused(parse(details.url).hostname)`

Reply:
Reference: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webNavigation/onCommitted#url